tf_fastmri_data/dataset_builder.py

```diff
@@ -99,10 +114,14 @@ def _get_path_default(self,):
         return path_default
 
     def _build_datasets(self):
+        if self.split_slices:
+            self.files_ds = self.files_ds.map(
+                lambda x : (x[0], int(x[1]))
+            )
         self._raw_ds = self.files_ds.map(
             partial(
                 load_data_from_file,
-                slice_random=self.slice_random,
+                slice_random=self.slice_random or self.split_slices,
```

Comment:
I think we should then change the name of this argument, because it does not reflect what's actually going on.
Maybe just another argument would be nice.

Reply:
Actually you don't even need this argument, you can always use the fact that `slice_index > -1` instead at run time.

Reply:
Wait a second, we need to tell early on that we are taking slices, so that there is no parsing issues. The code otherwise early on forms graph assuming that we may or may not take with slices, due to which the shape is drastically changed. I exactly faced this, hence I resorted to sending that slice_random is `True`:

```
TypeError: 'kspace_shape' must have the same nested structure in the main and else branches:
    
    The two structures don't have the same nested structure.
    
    First structure: type=tuple str=(None, None, None)
    
    Second structure: type=tuple str=(None, None, None, None)
    
```


What I propose is to rather use different argument or just call this argument as `select_slices`, which means we select slices, and if `slice_index = -1`, we choose at random. What do you think?